app/src/components/GrantDetailsRow.vue

```diff
@@ -0,0 +1,110 @@
+              </span>
+            </div>
+          </div>
+        </div>
+        <!-- add to cart, matching example -->
+        <div class="text-center xl:text-left col-span-12 mb-8 xl:col-span-5 xl:col-end-12">
+          <div class="px-4 xl:px-0 xl:text-right mb-8 xl:mb-0">
+            <div class="flex justify-center mb-6 xl:justify-end">
+              <!-- when item in cart add class "in-cart" + text in span "remove from cart"-->
+              <button v-if="isInCart(grant?.id)" @click="removeFromCart(grant?.id)" class="btn btn-primary">
+                <CartIcon class="icon-small" />
+                Remove from Cart
+              </button>
+              <button v-else @click="addToCart(grant?.id)" class="btn btn-primary">
+                <CartIcon class="icon-small" />
+                Add to Cart
+              </button>
+            </div>
+
+            <div>
+              <div>
+                10 {{ roundDetails[0] ? roundDetails[0].donationToken.symbol : '' }} ≈
+                <span v-for="(round, index) in roundDetails" :key="index">
+                  {{ round.prediction10 }} {{ round.matchingToken.symbol
+                  }}<span v-if="index + 1 < roundDetails.length">, </span>
+                </span>
+                Matching
+              </div>
+              <div>
+                100 {{ roundDetails[0] ? roundDetails[0].donationToken.symbol : '' }} ≈
+                <span v-for="(round, index) in roundDetails" :key="index">
+                  {{ round.prediction100 }} {{ round.matchingToken.symbol
+                  }}<span v-if="index + 1 < roundDetails.length">, </span>
+                </span>
+                Matching
+              </div>
+            </div>
+            <div class="flex justify-center mb-6 xl:justify-end">
+              <!-- allow extra buttons to be defined in the parent -->
+              <slot name="extraButtons"></slot>
+            </div>
+          </div>
+        </div>
+        <div class="text-left col-span-11 col-start-2">
+          <div class="flex flex-wrap gap-x-6 gap-y-4">
+            <div class="flex items-center gap-x-2 cursor-pointer group">
+              <ShareIcon class="icon-primary stroke-2 w-9" />
+              <span class="text-grey-400 group-hover:text-grey-500">Share</span>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue';
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+import { ArrowToprightIcon as ShareIcon } from '@fusion-icons/vue/interface';
+import { formatAddress } from 'src/utils/utils';
+import useCartStore from 'src/store/cart';
+import { GrantsRoundDetails } from '@dgrants/types';
+
+export default defineComponent({
+  name: 'GrantDetailsRow',
+  components: { CartIcon, ShareIcon },
+  props: {
+    grant: { type: Object, required: false, default: () => ({}) },
```

Comment:
Same comment that we should use a more specific prop type, might be others I missed too